src/dom.js

```diff
@@ -29,17 +29,22 @@ class VirtualRoot /*::<model, action>*/ {
     this.address = address
   }
   renderWith(current/*:Driver*/) {
+    let exception = null
     const previous = driver
     driver = current
 
     try {
       driver.render(this.view(this.model, this.address))
     }
     catch(error) {
-      driver = previous
-      throw error
+      exception = error
     }
+
     driver = previous
+
+    if (exception != null) {
```

Comment:
What if `render()` throws `null`? Probably not a problem in practice, but at least comment to that effect. If it could be a real problem, then change this to something like

```js
// Somewhere above `renderWith`
const NoExceptionSentinel = {};

// At the top of `renderWith`
let exception = NoExceptionSentinel;

// At the end of `renderWith`
if (exception !== NoExceptionSentinel) {
  ...
}
```